library/src/main/java/com/rengwuxian/materialedittext/TypefaceCache.java

```diff
@@ -0,0 +1,20 @@
+package com.rengwuxian.materialedittext;
+
+import android.content.res.AssetManager;
+import android.graphics.Typeface;
+
+import java.util.Hashtable;
+
+public class TypefaceCache {
+  private static final Hashtable<String, Typeface> FONT_CACHE = new Hashtable<>();
+
+  public static Typeface get(String name, AssetManager assets) {
+    synchronized (FONT_CACHE) {
```

Comment:
You`re right in case of synchronization. But I'm quite skeptical about weak references to typefaces. In situation you're moving from one activity to another, the fonts will be (probably/maybe) garbage collected and loaded again. It is expected the font will be changed in whole application if it's changed in EditTexts. After all, It's not a big deal.